api/controllers/DashboardController.js

```diff
@@ -16,275 +14,360 @@ var dashboardController = {
-                        if (currentPage != 1) {
-                            return res.view('404.ejs');
-                        }
-                        else {
-                            message = 'No results found.';
-                        }
-                    }
-                    else {
-                        resultCount = results[0].result_count;
-                    }
-
-                    async.parallel ([
-                        // Make call to Casebook Status API for the application references in the results collection.
-
-                        function(callback){
-
-                            var request = require('request');
-                            var fs = require('fs');
-                            var crypto = require('crypto');
-                            var apiQueryString = require('querystring');
-
-                            // Create status retrieval request object.
-
-                            // First build array of application references to be passed to the Casebook Status API for this page. Can submit up to 20 at a time.
-
-                            var applicationReferences = [];
-
-                            for ( var i=0; i < results.length; i++ ) {
-                                applicationReferences.push(results[i].unique_app_id);
-                            }
+                    type: sequelize.QueryTypes.SELECT,
+                };
+                const userData = HelperService.getUserData(req, res);
+                const displayAppsArgs = {
+                    userData,
+                    totalApplications,
+                    offset,
+                    sortOrder,
+                    currentPage,
+                    searchCriteria,
+                    pageSize,
+                    req,
+                    res,
+                };
+
+                dashboardController._getApplications(
+                    storedProcedureArgs,
+                    displayAppsArgs,
+                    userData.user.electronicEnabled
+                );
+            });
+        });
+    },
 
-                            // Create Request Structure
+    _getApplications(storedProcedureArgs, displayAppsArgs, electronicEnabled) {
+        const applicationType = electronicEnabled
+            ? 'electronic and paper'
+            : 'paper';
+        const queryApplications = dashboardController._chooseStoredProcedure(
+            storedProcedureArgs.secondarySortOrder,
+            electronicEnabled
+        );
+
+        sails.log.info(`Fetching ${applicationType} applications`);
+        sequelize
+            .query(queryApplications, storedProcedureArgs)
+            .then((electronitApplications) => {
+                dashboardController._displayApplications(
+                    electronitApplications,
+                    displayAppsArgs
+                );
+            })
+            .catch((error) => {
+                sails.log(error);
```

Comment:
I'm not sure I follow, how did you figure out it's a silent error? Is it because it's missing a log method?
`sails.log.error(error);` instead of `sails.log(error);`

Reply:
silent wasn't the best term sorry 😄, I meant if just logging the error is OK - and does not affect the rest of the flow of the code/there's no need to "recover" from the code you can ignore this. (for example redirecting the user to an error page etc)